mantis-control-plane/mantis-control-plane-server/src/main/java/io/mantisrx/server/master/persistence/MantisStorageProviderAdapter.java

```diff
@@ -50,6 +66,27 @@
 public class MantisStorageProviderAdapter implements IMantisStorageProvider {
```

Comment:
What do you think about making this an abstract class and getting rid of the MantisStorageProvider interface (I'm referring to the kvdal methods such as get, upsert, etc...) methods and moving them here?

Reply:
I think that works better. In fact, I attempted down this road but it becomes a slightly bigger refactor in how the instance of this storage class is initialized here in a hard-coded whereas we create the instance of underlying `MantisStorageProvider` using reflection in a config driven manner.

How about I take a stab at it in a subsequent pass?